substrate/frame/message-queue/src/lib.rs

```diff
@@ -759,6 +760,47 @@ impl<T: Config> Pallet<T> {
 		}
 	}
 
+	/// The maximal weight that a single message can consume.
+	///
+	/// Any message using more than this will be marked as permanently overweight and not
+	/// automatically re-attempted. Returns `None` if the servicing of a message cannot begin.
+	/// `Some(0)` means that only messages with no weight may be served.
+	fn overweight_limit(limit: Weight) -> Option<Weight> {
```

Comment:
Maybe it's just me, but I'd find it less confusing if `overweight_limit` became `max_message_weight`, so we don't have two limits. WDYT?

Reply:
Yea agreed. It sounds confusing :see_no_evil: